libs/langchain/langchain/chains/synthetic_data/__init__.py

```diff
@@ -0,0 +1,47 @@
+from typing import Any, Dict, List, Optional
+
+from langchain.chains.base import Chain
+from langchain.chains.llm import LLMChain
+from langchain.chains.synthetic_data.prompts import SENTENCE_PROMPT
+from langchain.prompts import PromptTemplate
```

Comment:
@piotr-grodek-dsai @maks-operlejn-ds @mateusz-wosinski-ds 
if `PromptTemplate` and `BaseLanguageModel` are imported only for type hints, is it worth to import `TYPE_CHECKING` from `typing` and add 
```
if TYPE_CHECKING:
    ...
```
logic here ?

I've encountered this suggestion from LangChain team during my PR, so maybe it's good practice that I didn't know

Reply:
Yes, if something is required only for typing then it should be imported under if.